src/api/routes/controllers/chats.controller.js

```diff
@@ -0,0 +1,68 @@
+const jwt = require("jsonwebtoken");
+const chatsService = require("../../services/chats.service");
+
+exports.getChat = async (req, res, next) => {
+  const { id } = req.params;
+  const { lastIndex } = res.locals;
+  const DEFAULT_LENGTH = 30;
+
+  try {
+    const response = await chatsService.getChatsFromGoalId(id);
+
+    if (!lastIndex) {
+      const rangeStart =
+        response.length < DEFAULT_LENGTH ? 0 : response.length - DEFAULT_LENGTH;
+      const chats = response.slice(rangeStart, response.length);
+      const newNextPageToken =
+        rangeStart === 0
+          ? null
+          : jwt.sign(
+              {
+                lastIndex: rangeStart,
+              },
+              process.env.CHAT_QUERY_SECRET_KEY,
+              {
+                expiresIn: "1d",
+              },
+            );
+
+      return res.json({
+        result: {
+          messages: chats,
+          nextPageToken: newNextPageToken,
+        },
+      });
+    }
+
+    const rangeStart =
+      lastIndex < DEFAULT_LENGTH ? 0 : lastIndex - DEFAULT_LENGTH;
+    const chats = response.slice(rangeStart, lastIndex);
+    const newNextPageToken =
+      rangeStart === 0
+        ? null
+        : jwt.sign(
+            {
+              lastIndex: rangeStart,
+            },
+            process.env.CHAT_QUERY_SECRET_KEY,
+            {
+              expiresIn: "1d",
+            },
+          );
+
+    return res.json({
+      result: {
+        messages: chats,
+        nextPageToken: newNextPageToken,
+      },
+    });
+  } catch (error) {
+    res.json({
+      result: "error",
+      error: {
+        message: "messages not found",
+        code: 400,
+      },
+    });
+  }
```

Comment:
```suggestion
      res.status(400);
      res.json({
        result: "error",
        message: "messages not found",
      });
  }
```
에러 처리 잘 해주셨네요👍
이전에 통일한 스타일에 맞추어 작성해주셔도 좋을 것 같습니다!